linera-execution/src/evm/revm.rs

```diff
@@ -664,20 +677,27 @@ where
 }
 
 fn process_execution_result(
+    storage_stats: StorageStats,
     result: ExecutionResult,
 ) -> Result<ExecutionResultSuccess, ExecutionError> {
     match result {
         ExecutionResult::Success {
             reason,
-            gas_used: _,
-            gas_refunded: _,
+            gas_used,
+            gas_refunded,
             logs,
             output,
-        } => Ok(ExecutionResultSuccess {
-            reason,
-            logs,
-            output,
-        }),
+        } => {
+            let mut gas_final = gas_used;
+            gas_final -= storage_stats.storage_costs();
```

Comment:
Maybe checked_sub here?

Reply:
I think no.
Checked sub is about doing subtraction that can fail and then induce an error. This is the stuff that happens if you do not enough fuel for running your application.

However, in that case, you are subtracting the storage costs from the total gas. So, the total costs should be higher than the storage costs. If not then you have an error. That error creating a panic is in fact appropriate.

Reply:
Yes, I understand. I just don't think panicing here is better than raising an appropriate error.